libs/domains/clusters/feature/src/lib/cluster-action-toolbar/cluster-action-toolbar.tsx

```diff
@@ -0,0 +1,231 @@
+import { useCopyToClipboard } from '@qovery/shared/util-hooks'
+import {
+  isDeleteAvailable,
+  isDeployAvailable,
+  isRedeployAvailable,
+  isStopAvailable,
+  isUpdateAvailable,
+} from '@qovery/shared/util-js'
+import { ClusterDeleteModal } from '../cluster-delete-modal/cluster-delete-modal'
+import { useDeployCluster } from '../hooks/use-deploy-cluster/use-deploy-cluster'
+import { useDownloadKubeconfig } from '../hooks/use-download-kubeconfig/use-download-kubeconfig'
+import { useStopCluster } from '../hooks/use-stop-cluster/use-stop-cluster'
+
+function MenuManageDeployment({
+  cluster,
+  clusterStatus,
+  organizationId,
+}: {
+  cluster: Cluster
+  clusterStatus: ClusterStatusGet
+  organizationId: string
+}) {
+  const { openModalConfirmation } = useModalConfirmation()
+  const { mutate: deployCluster } = useDeployCluster()
+  const { mutate: stopCluster } = useStopCluster()
+
+  if (
+    !clusterStatus.status ||
+    (!isDeployAvailable(clusterStatus?.status) &&
+      !isDeleteAvailable(clusterStatus?.status) &&
+      !isUpdateAvailable(clusterStatus?.status))
+  ) {
+    return null
+  }
+
+  const mutationDeploy = () =>
+    deployCluster({
+      organizationId,
+      clusterId: cluster.id,
+    })
+  const mutationUpdate = () =>
+    openModalConfirmation({
+      mode: EnvironmentModeEnum.PRODUCTION,
+      title: 'Confirm update',
+      description: 'To confirm the update of your cluster, please type the name:',
+      name: cluster.name,
+      action: () =>
+        deployCluster({
+          organizationId,
+          clusterId: cluster.id,
+        }),
+    })
+  const mutationStop = () =>
+    openModalConfirmation({
+      mode: EnvironmentModeEnum.PRODUCTION,
+      title: 'Confirm stop',
+      description: 'To confirm the stop of your cluster, please type the name:',
+      warning:
+        'Please note that by stopping your cluster, some resources will still be used on your cloud provider account and still be added to your bill. To completely remove them, please use the “Remove” feature.',
+      name: cluster.name,
+      action: () =>
+        stopCluster({
+          organizationId,
+          clusterId: cluster.id,
+        }),
+    })
+
+  const entries: ReactNode[] = [
+    isDeployAvailable(clusterStatus.status) && (
+      <DropdownMenu.Item key="0" icon={<Icon name={IconAwesomeEnum.PLAY} />} onClick={mutationDeploy}>
+        {clusterStatus.is_deployed ? 'Deploy' : 'Install'}
+      </DropdownMenu.Item>
+    ),
+    isRedeployAvailable(clusterStatus.status) && (
+      <DropdownMenu.Item key="1" icon={<Icon name={IconAwesomeEnum.ROTATE_RIGHT} />} onClick={mutationUpdate}>
+        Update
+      </DropdownMenu.Item>
+    ),
+    cluster.cloud_provider !== 'GCP' && isStopAvailable(clusterStatus.status) && (
+      <DropdownMenu.Item key="2" icon={<Icon name={IconAwesomeEnum.CIRCLE_STOP} />} onClick={mutationStop}>
+        Stop
+      </DropdownMenu.Item>
+    ),
+  ].filter((e) => !!e)
```

Comment:
I think it's a more clean and lisible to have directly a condition without `map`, it allow to remove `filter` and generic `key`

if it's mainly for the `return entries.length > 0 ? (` we have already the condition that return `null` on the top